packages/engine/src/simulation/package/run.rs

```diff
@@ -160,18 +160,18 @@ impl StepPackages {
         let snapshot_arc = Arc::new(snapshot);
 
         pkgs.into_iter().for_each(|mut package| {
-            let state_arc = state.clone();
+            let state_clone = state.clone();
```

Comment:
Honestly I find this a little confusing. I know in an IDE you have type-hints and such, but the naming of the variable makes it feel like you've cloned the actual state (rather than the arc to the state)

Reply:
Can we not just shadow it and say `let state = state.clone()`

Reply:
Sorry, I think this got merged by accident. I agree with shadowing.